types/utils.go

```diff
@@ -91,7 +91,12 @@ func NewLevelDB(name, dir string) (db dbm.DB, err error) {
 			err = fmt.Errorf("couldn't create db: %v", r)
 		}
 	}()
-	return dbm.NewDB(name, backend, dir), err
+
+	if err != nil {
```

Comment:
is this necessary? won't the error be returned because the `err error` definition in the return

Reply:
I think a direct return of the error in the defer func makes the most sense